File: Algorithm/src/Main_1260.java
import java.io.*;
import java.util.*;

public class Main_1260 {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static StringTokenizer st;
    static List<List<Integer>> graph = new ArrayList<>();
    static boolean[] visited;

    public static void main(String[] args) throws IOException {
        st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        int V = Integer.parseInt(st.nextToken());

        for (int i = 0; i < N + 1; i++) {
            graph.add(new ArrayList<>());
        }

        for (int i = 0; i < M; i++) {
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());

            graph.get(a).add(b);
            graph.get(b).add(a);
        }

        for (int i = 1; i < N + 1; i++) {
            Collections.sort(graph.get(i));
        }
        visited = new boolean[N + 1];
        dfs(V);
        bw.write("\n" + "");
        visited = new boolean[N + 1];
        bfs(V);
        bw.close();
    }

    static void bfs(int start) throws IOException {
        bw.write(start + " ");
        Queue<Integer> queue = new ArrayDeque<>();
        visited[start] = true;
        queue.add(start);

        while (!queue.isEmpty()) {
            int cur = queue.poll();

            for (Integer ele : graph.get(cur)) {
                if (visited[ele]) continue;
                queue.add(ele);
                visited[ele] = true;
                bw.write(ele + " ");
            }
        }
    }

    static void dfs(int start) throws IOException {
        if (visited[start]) return;

        bw.write(start + " ");
        visited[start] = true;

        for (Integer ele : graph.get(start)) {
            if (visited[ele]) continue;
            dfs(ele);
        }
    }
}

File: Algorithm/python/Main_1260.py
import collections
import sys

input = lambda: sys.stdin.readline().rstrip()

N, M, V = map(int, input().split())
graph = collections.defaultdict(list)
visited = collections.defaultdict(lambda: False)

for i in range(M):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)


def dfs(v):
    visited[v] = True
    print(v, end=' ')
    for i in sorted(graph[v]):
        if not visited[i]:
            dfs(i)


def bfs(v):
    q = collections.deque()
    visited = collections.defaultdict(lambda: False)
    q.append((v))
    print(v, end=' ')
    visited[v] = True

    while q:
        cur = q.popleft()
        for i in sorted(graph[cur]):
            if not visited[i]:
                visited[i] = True
                print(i, end=' ')
                q.append(i)


dfs(V)
print()
bfs(V)
